docs/declaring-loaders.rst

```diff
@@ -5,22 +5,41 @@ Declaring Item Loaders
 
 Item Loaders are declared by using a class definition syntax. Here is an example::
 
+    from dataclasses import dataclass, field
+    from typing import Optional
+
     from itemloaders import ItemLoader
-    from itemloaders.processors import TakeFirst, MapCompose, Join
+    from itemloaders.processors import TakeFirst
 
-    class ProductLoader(ItemLoader):
 
-        default_output_processor = TakeFirst()
+    def price_out(values):
+        return float(values[0])
+
+
+    @dataclass
+    class Product:
+        name: Optional[str] = field(default=None)
+        price: Optional[float] = field(
+            default=None,
+            metadata={'output_processor': price_out}
+        )
 
-        name_in = MapCompose(str.title)
-        name_out = Join()
+
+    class ProductLoader(ItemLoader):
+        default_item_class = Product
 
         # using a built-in processor
-        price_in = MapCompose(str.strip)
+        default_output_processor = TakeFirst()
 
         # using a function
-        def price_out(self, values):
-            return float(values[0])
+        def name_in(self, values):
+            return values[0].title()
+
+    loader = ProductLoader()
+    loader.add_value('name', 'plasma TV')
+    loader.add_value('price', '999.98')
+    loader.load_item()
+    # Product(name='Plasma Tv', price=999.98)
```

Comment:
Do we want this example to feature dataclass? Shouldn’t it use `Item`, or even `dict`, to keep the example as simple as possible?

That would, of course, mean creating a separate section about using dataclasses, probably with the code from the docstring warning below.

Reply:
My thought here is that, this is the easiest way to show `metadata`.
Other than that, there is no simple way, unless we write plain text.
`dict` doesn't support it.
`Item` requires `scrapy`, but this library is independent of scrapy.
I can set it there, but it wouldn't be self-contained.
dataclasses may have the same problem, but as they are from python standard library, seems more natural.

Reply:
Makes sense.